### PR TITLE
Migrate FFI to ES modules

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

Migrates FFI to ES modules

---

**Checklist:**

- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: src/Effect/Ref.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
 "use strict";
 
-exports.new = function (val) {
+export var _new = function (val) {
```
Comment (thomashoneyman) [inline src/Effect/Ref.js:3]: I thought we'd settled on `export const ...`; is this simply an old commit that should be updated, or have we switched to `var`?

**Comment by thomashoneyman** [comment]
We're also going to have to update our ESLint configurations to no longer be commonjs:

https://github.com/purescript/purescript-refs/blob/f66d3cdf6a6bf4510e5181b3fac215054d8f1e2e/.eslintrc.json#L6-L8

File: src/Effect/Ref.js (hunk 1 above)
Comment (JordanMartinez) [reply src/Effect/Ref.js:3]: At first, I was just opening up the PRs that were being used by the ES modules PR without any additional modifications. Then, I got done with 'core' libraries and worked on repos that didn't have that work already done. So, I started using `lebab` and some replacement stuff.

So, we still are going to switch  to `export const`. This PR just needs to be updated to have that.

**Comment by JordanMartinez** [comment]
> We're also going to have to update our ESLint configurations to no longer be commonjs:

What should the new value be? [There's a couple of possible options](https://eslint.org/docs/user-guide/configuring/language-options#specifying-environments).

Looks like this will do the trick:
```sh
 $ jq 'del(.env.commonjs) | .env |= { foo: "bar" }' .eslintrc.json
 {
  "parserOptions": {
    "ecmaVersion": 5
  },
  "extends": "eslint:recommended",
  "env": {
    "foo": "bar"
  },
  "rules": {
    "strict": [
      2,
      "global"
    ],
    "block-scoped-var": 2,
    "consistent-return": 2,
    "eqeqeq": [
      2,
      "smart"
    ],
    "guard-for-in": 2,
    "no-caller": 2,
    "no-extend-native": 2,
    "no-loop-func": 2,
    "no-new": 2,
    "no-param-reassign": 2,
    "no-return-assign": 2,
    "no-unused-expressions": 2,
    "no-use-before-define": 2,
    "radix": [
      2,
      "always"
    ],
    "indent": [
      2,
      2
    ],
    "quotes": [
      2,
      "double"
    ],
    "semi": [
      2,
      "always"
    ]
  }
}
```

**Comment by thomashoneyman** [comment]
I think we need to do two things:

1. If there is an `"env"` key with multiple entries, delete the `"commonjs"` entry. If it's the only entry, delete the `"env"` key altogether.
2. If there is a `"parserOptions"` key, and it has a `"sourceType"` entry, then update that entry to be `"module"`. Otherwise, add the `"sourceType"` entry to the `"parserOptions"` key. And if there is no `"parserOptions"` key, then add it, and add the `"sourceType"` entry.

More details on the parser:
https://eslint.org/docs/user-guide/configuring/language-options#specifying-parser-options

**Pull request closed without merging**